### PR TITLE
Clean up a few more bits of verifiable maps.

### PR DESCRIPTION
**Pull request by pphaneuf**
### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: trillian.proto [hunk 1]
```diff
@@ -99,15 +99,15 @@ enum TreeType {
   // Tree represents a verifiable log.
   LOG = 1;
 
-  // Tree represents a verifiable map.
-  MAP = 2;
+  // Deprecated: verifiable maps are no longer supported.
+  DEPRECATED_MAP = 2 [deprecated = true];
```
Comment (Martin2112) [inline trillian.proto:103]: Not sure there's any point deprecating it as it already no longer works. Just reserve the field?

Comment (pphaneuf) [reply trillian.proto]: Good call, done.

File: trillian.proto [hunk 2]
```diff
@@ -99,15 +99,15 @@ enum TreeType {
...
 
-  // Tree represents a verifiable map.
-  MAP = 2;
-
   // Tree represents a verifiable pre-ordered log, i.e., a log whose entries are
   // placed according to sequence numbers assigned outside of Trillian.
   PREORDERED_LOG = 3;
+
+  reserved 2;
+  reserved "MAP";
```
Comment (pavelkalinnikov) [inline trillian.proto:107]: I wonder whether reserving these has any effect in Go. The diff in ".pb.go" file doesn't seem different from just deleting these lines?

**Review by pavelkalinnikov — APPROVED** [review]
LGTM. One question inlined.

File: trillian.proto (hunk 2 above)
Comment (pavelkalinnikov) [reply trillian.proto:107]: Ah, this looks like not a compatibility mechanism, but it just prevents future edits reusing this field number/name?

Comment (pphaneuf) [reply trillian.proto:107]: Yeah, they're the same, their effect is in `protoc`, where if you try to reuse the name or value, it'll throw an error.

This used to be done by just leaving a comment, traditionally, but sometimes, you have these messages that are very long, and people miss the comment...

**Review by Martin2112 — APPROVED** [review]

**Pull request merged**